dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/utils/MemoryTransformer.java

```diff
@@ -0,0 +1,124 @@
+package com.dat3m.dartagnan.parsers.program.visitors.spirv.utils;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.processing.ExprTransformer;
+import com.dat3m.dartagnan.parsers.program.visitors.spirv.decorations.BuiltIn;
+import com.dat3m.dartagnan.program.Function;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.Thread;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.memory.MemoryObject;
+import com.dat3m.dartagnan.program.memory.ScopedPointerVariable;
+import com.dat3m.dartagnan.program.memory.VirtualMemoryObject;
+import com.dat3m.dartagnan.program.misc.NonDetValue;
+
+import java.util.*;
+import java.util.function.IntUnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+
+public class MemoryTransformer extends ExprTransformer {
+
+    private static final List<String> namePrefixes = List.of("T", "S", "W", "Q", "D");
```

Comment:
I would prefer to avoid unnecessary verbose comments. I think it should be clear from the code that they are thread, subgroup, workgroup, queue family, and device.

Reply:
ok